examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1520 @@
+    "The following class loads the data and transforms it in the way it is required for resnet50. It is written in the form such that it is compatible with torch.utils.data.DataLoader. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class mimicDataset(Dataset):\n",
+    "    \n",
+    "    def __init__(self, path):\n",
+    "        'Initialization'\n",
+    "        self.path = path\n",
+    "        #self.y = y\n",
+    "        \n",
+    "    def __len__(self):\n",
+    "        'Denotes the total number of samples'\n",
+    "        return len(self.path)\n",
+    "    \n",
+    "    def __getitem__(self, index):\n",
+    "        'Generates one sample of data'\n",
+    "        # Select sample\n",
+    "        image = Image.open(\"\".join([\"physionet.org/files/mimic-cxr-jpg/2.0.0/\",self.path[index,3].replace(\".dcm\", \".jpg\")])).convert('RGB')\n",
+    "        X = self.transform(image)\n",
+    "        label = self.path[index,6]\n",
+    "        \n",
+    "        return X, torch.tensor(label)\n",
+    "    \n",
+    "    transform = transforms.Compose([\n",
+    "        transforms.Resize(256),\n",
+    "        transforms.CenterCrop(224),\n",
+    "        transforms.ToTensor(),\n",
+    "        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We load the resnet50 model remove the last layer, so the output is then the second last layer with dimension 1x2048.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['all_X.lib']"
+      ]
+     },
+     "execution_count": 28,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model = models.resnet50(pretrained=True)\n",
+    "modules = list(model.children())[:-1]\n",
+    "model=torch.nn.Sequential(*modules)\n",
+    "for p in model.parameters():\n",
+    "    p.requires_grad = False\n",
+    "    \n",
+    "model.eval()\n",
+    "#apply modified resnet50 to data\n",
+    "dataloaders = DataLoader(mimicDataset(input_list[:n,:]), batch_size=n,num_workers=0)\n",
+    "    \n",
+    "data, labels = next(iter(dataloaders))\n",
+    "with torch.no_grad():\n",
+    "    features_var = model(data)\n",
+    "    features = features_var.data \n",
+    "    all_X = features.reshape(n,2048).numpy()\n",
+    "    \n",
+    "#save feature matrix\n",
+    "dump(all_X, \"all_X.lib\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finetuning a pretrained CNN and extracting the second last layer as features"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the second approach of image encoding, we use the concept of transfer learning. \n",
```

Comment:
I missed the first one :D

Reply:
Ah ok, I added 'first approach' and 'second approach' in the headings, so it should be clearer now. Basically in the first approach I don't do any finetuning and in the second I finetune the model with the Chexpert labels. 

The reason for the two approaches is: The second approach uses the Chexpert labels for finetuning and I was not sure if we want to touch our labels there since this might be confusing with our goal to demonstrate weak supervision. 
I think it depends whether we evaluate our model with gold labels in the end or (if we don't find any appropriate gold labels) we have to stick with the Chexpert labels.  In the latter case, I would not use the Chexpert labels for the image encoding, but rather go with the first approach (without finetuning).
Alternatively I could finetune the model for the image encoding on different data, like  iuxray.

Reply:
Since we have a test set from the chexpert data, the finetuning with the weak labels sounds fine for me. But I think I will leave both approaches and compare the results in the end.